src/puya/awst_build/eb/temporary_assignment.py

```diff
@@ -11,14 +12,20 @@
 from puya.parse import SourceLocation
 
 
+# TODO: this should probably extend ValueProxyExpressionBuilder
```

Comment:
Can we do this in this PR, or create an issue to track it rather than adding more todo comments

Reply:
I've fixed the actual problem issues in this PR - namely that some places were checking for instances of `ValueProxyExpressionBuilder` instead of looking at `value_type`, and also made this class properly implement said property. 
I don't want to make this PR any bigger by doing this additional refactoring, and I don't think it's worthy of an issue, so I'm just gonna change the comment from a TODO to a REFACTOR :P